src/uhttp.c

```diff
@@ -1058,65 +1064,75 @@ HTTP_CLIENT_RESULT uhttp_client_open(HTTP_CLIENT_HANDLE handle, const char* host
             result = HTTP_CLIENT_INVALID_STATE;
         }
-        else if (mallocAndStrcpy_s(&http_data->host_name, host) != 0)
-        {
-            LogError("copying hostname has failed");
-            result = HTTP_CLIENT_ERROR;
-        }
-        /* Codes_SRS_UHTTP_07_007: [http_client_connect shall attempt to open the xio_handle. ] */
         else
         {
-            result = HTTP_CLIENT_OK;
-            http_data->recv_msg.recv_state = state_opening;
-            http_data->on_connect = on_connect;
-            http_data->connect_user_ctx = callback_ctx;
-            http_data->port_num = port_num;
+            if (http_data->host_name != NULL)
+            {
+                free(http_data->host_name);
+                handle->host_name = NULL;
+            }
 
-            if (http_data->x509_cert != NULL && http_data->x509_pk != NULL)
+            if (mallocAndStrcpy_s(&http_data->host_name, host) != 0)
             {
-                if (xio_setoption(http_data->xio_handle, SU_OPTION_X509_CERT, http_data->x509_cert) != 0 || xio_setoption(http_data->xio_handle, SU_OPTION_X509_PRIVATE_KEY, http_data->x509_pk) != 0)
-                {
-                    LogError("Failed setting x509 certificate");
-                    result = HTTP_CLIENT_ERROR;
-                    free(http_data->host_name);
-                    http_data->host_name = NULL;
-                    http_data->on_connect = NULL;
-                    http_data->connect_user_ctx = NULL;
-                    http_data->port_num = 0;
-                }
+                LogError("copying hostname has failed");
+                result = HTTP_CLIENT_ERROR;
             }
-            if (result == HTTP_CLIENT_OK && http_data->certificate != NULL)
+            /* Codes_SRS_UHTTP_07_007: [http_client_connect shall attempt to open the xio_handle. ] */
+            else
             {
-                if (xio_setoption(http_data->xio_handle, OPTION_TRUSTED_CERT, http_data->certificate) != 0)
+                result = HTTP_CLIENT_OK;
+                http_data->recv_msg.recv_state = state_opening;
+                http_data->on_connect = on_connect;
+                http_data->connect_user_ctx = callback_ctx;
+                http_data->port_num = port_num;
+
+                if (http_data->x509_cert != NULL && http_data->x509_pk != NULL)
                 {
-                    LogError("Failed setting Trusted certificate");
-                    result = HTTP_CLIENT_ERROR;
-                    free(http_data->host_name);
-                    http_data->host_name = NULL;
-                    http_data->on_connect = NULL;
-                    http_data->connect_user_ctx = NULL;
-                    http_data->port_num = 0;
+                    if (xio_setoption(http_data->xio_handle, SU_OPTION_X509_CERT, http_data->x509_cert) != 0 || xio_setoption(http_data->xio_handle, SU_OPTION_X509_PRIVATE_KEY, http_data->x509_pk) != 0)
```

Comment:
>xio_setoption(http_data->xio_handle, SU_OPTION_X509_CERT, http_data->x509_cert) != 0 [](start = 24, length = 84)

may be just personal preference but i prefer to have the conditions grouped with parenthesis. makes more explicit intended evaluation.

Reply:
I like the same - there are too many in the file and don't want to make that many changes. These lines just got indented when I've added one new condition.
We may want to re-think our coding guidelines for error-checking conditions to avoid needing changing most of function if a new statement is added.  (@jspaith , @rafilho FYI).

Reply:
yea i mean in general i think we can iteratively make things better. so if anyone goes in to fix something for a PR and there are some style improvements that can be made with the things you are specifically touching i think they should be made.